cocotb/triggers.py

```diff
@@ -934,7 +940,12 @@ def __repr__(self):
         return fmt.format(type(self).__qualname__, self.signal, self.num_cycles)
 
 
-async def with_timeout(trigger, timeout_time, timeout_unit="step"):
+async def with_timeout(
+    trigger: Union[Trigger, Waitable, Coroutine[Any, Any, T]],
+    timeout_time: float,
```

Comment:
```suggestion
    timeout_time: Union[numbers.Real, decimal.Decimal],
```
?

Reply:
Technically this doesn't work with type checking. No type checker supports the numeric tower right now. Maybe
```suggestion
    timeout_time: Union[float, decimal.Decimal],
```